<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1145</width>
    <height>870</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Unlimited Blade Works</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="proGLWidget" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>800</width>
      <height>800</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>670</y>
      <width>321</width>
      <height>71</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QPushButton" name="iniButton">
       <property name="text">
        <string>初始化</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QPushButton" name="stopButton">
       <property name="text">
        <string>停止</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="startButton">
       <property name="text">
        <string>开始</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="saveButton">
       <property name="text">
        <string>保存状态</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QPushButton" name="revButton">
       <property name="text">
        <string>反转</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="loadButton">
       <property name="text">
        <string>读取状态</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>780</y>
      <width>321</width>
      <height>22</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="Label_t">
         <property name="text">
          <string>T=</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="timeLabel">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="label_FPS">
         <property name="text">
          <string>FPS:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="fpsLabel">
         <property name="text">
          <string>---</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="groupBox_1">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>10</y>
      <width>321</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>Gas:1</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>111</width>
       <height>53</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="G1_N">
          <property name="text">
           <string>数量：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="G1_num">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="G1_T">
          <property name="text">
           <string>温度：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="G1_temp">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>120</y>
      <width>321</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>Gas:2</string>
    </property>
    <widget class="QWidget" name="layoutWidget_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>111</width>
       <height>53</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="G2_N">
          <property name="text">
           <string>数量：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="G2_num">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QLabel" name="G2_T">
          <property name="text">
           <string>温度：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="G2_temp">
          <property name="text">
           <string>0</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_5">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <zorder>layoutWidget</zorder>
   <zorder>openGLWidget</zorder>
   <zorder>gridLayoutWidget</zorder>
   <zorder>groupBox_1</zorder>
   <zorder>groupBox_2</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1145</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>帮助</string>
    </property>
   </widget>
   <addaction name="menu"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionK">
   <property name="text">
    <string/>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>proGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>proglwidget.h</header>
   <slots>
    <signal>timeCost(int)</signal>
    <slot>initialize(int)</slot>
    <slot>step()</slot>
    <slot>reverse()</slot>
    <slot>initialize_LR(int,int)</slot>
    <slot>step(int)</slot>
    <slot>initialize(int,double,double,double,int,int,double,double,double,int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>iniButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>iniB()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>889</x>
     <y>723</y>
    </hint>
    <hint type="destinationlabel">
     <x>889</x>
     <y>688</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>startButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>startB()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>861</x>
     <y>769</y>
    </hint>
    <hint type="destinationlabel">
     <x>864</x>
     <y>815</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stopButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>stopB()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>967</x>
     <y>772</y>
    </hint>
    <hint type="destinationlabel">
     <x>972</x>
     <y>817</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>revButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>revB()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1069</x>
     <y>762</y>
    </hint>
    <hint type="destinationlabel">
     <x>1052</x>
     <y>806</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>reverseVelo()</signal>
   <receiver>openGLWidget</receiver>
   <slot>reverse()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>703</x>
     <y>843</y>
    </hint>
    <hint type="destinationlabel">
     <x>698</x>
     <y>808</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>initialize_LR(int,int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>initialize_LR(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>907</x>
     <y>637</y>
    </hint>
    <hint type="destinationlabel">
     <x>716</x>
     <y>624</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>loadButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadB()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1000</x>
     <y>728</y>
    </hint>
    <hint type="destinationlabel">
     <x>997</x>
     <y>692</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveB()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1055</x>
     <y>724</y>
    </hint>
    <hint type="destinationlabel">
     <x>1047</x>
     <y>695</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>showFPS(int)</signal>
   <receiver>fpsLabel</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1110</x>
     <y>803</y>
    </hint>
    <hint type="destinationlabel">
     <x>1117</x>
     <y>832</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>showTime(QString)</signal>
   <receiver>fpsLabel</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>945</x>
     <y>834</y>
    </hint>
    <hint type="destinationlabel">
     <x>1116</x>
     <y>827</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>showTime(QString)</signal>
   <receiver>timeLabel</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>842</x>
     <y>806</y>
    </hint>
    <hint type="destinationlabel">
     <x>840</x>
     <y>828</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openGLWidget</sender>
   <signal>timeCost(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>timeEve(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>763</x>
     <y>663</y>
    </hint>
    <hint type="destinationlabel">
     <x>812</x>
     <y>665</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>timePass(int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>step(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>843</x>
     <y>487</y>
    </hint>
    <hint type="destinationlabel">
     <x>702</x>
     <y>510</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>timePass()</signal>
   <receiver>openGLWidget</receiver>
   <slot>step()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>863</x>
     <y>578</y>
    </hint>
    <hint type="destinationlabel">
     <x>718</x>
     <y>567</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>initialize(int,double,double,double,int,int,double,double,double,int)</signal>
   <receiver>openGLWidget</receiver>
   <slot>initialize(int,double,double,double,int,int,double,double,double,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>813</x>
     <y>639</y>
    </hint>
    <hint type="destinationlabel">
     <x>782</x>
     <y>643</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>setT1(QString)</signal>
   <receiver>G1_temp</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>851</x>
     <y>331</y>
    </hint>
    <hint type="destinationlabel">
     <x>877</x>
     <y>118</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>setT2(QString)</signal>
   <receiver>G2_temp</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>922</x>
     <y>375</y>
    </hint>
    <hint type="destinationlabel">
     <x>878</x>
     <y>215</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>setN1(int)</signal>
   <receiver>G1_num</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>914</x>
     <y>354</y>
    </hint>
    <hint type="destinationlabel">
     <x>875</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MainWindow</sender>
   <signal>setN2(int)</signal>
   <receiver>G2_num</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>838</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>877</x>
     <y>192</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>initialize_LR(int,int)</signal>
  <signal>timePass()</signal>
  <signal>reverseVelo()</signal>
  <signal>showFPS(int)</signal>
  <signal>showTime(QString)</signal>
  <signal>timePass(int)</signal>
  <signal>initialize(int,double,double,double,int,int,double,double,double,int)</signal>
  <signal>setT1(QString)</signal>
  <signal>setT2(QString)</signal>
  <signal>setN1(int)</signal>
  <signal>setN2(int)</signal>
  <slot>iniB()</slot>
  <slot>startB()</slot>
  <slot>stopB()</slot>
  <slot>loadB()</slot>
  <slot>saveB()</slot>
  <slot>revB()</slot>
  <slot>timeEve(int)</slot>
 </slots>
</ui>
